src/Authoring/WinRT.SourceGenerator/Generator.cs

```diff
@@ -269,6 +276,40 @@ public static int DllCanUnloadNow()
                 }
                 """);
         }
+
+        /// <summary>
+        /// Generates the native exports for a WinRT component.
+        /// </summary>
+        /// <param name="context">The <see cref="GeneratorExecutionContext"/> value to use to produce source files.</param>
+        public static void GenerateWinRTExportsType(GeneratorExecutionContext context)
+        {
+            if (context.Compilation.AssemblyName is not { Length: > 0 } assemblyName)
+            {
+                return;
+            }
+

```

Comment:
I recommend referring to the exact set of allowed characters in assembly vs namespace names and transforming assemblyName accordingly here. For example, '-' is valid in assembly names, but not namespace names, which has resulted in similar bugs before, e.g. https://github.com/OmniSharp/generator-aspnet/issues/256

Reply:
Yup, just asked @333Fred about this as well (he mentioned II.6.2 and II.5.2/3 of ECMA 335), I'll push a fix for this 👍